pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -60,6 +61,44 @@ func GetEnvironment(cr *api.KieApp, service kubernetes.PlatformService) (api.Env
 		cr.Status.Applied.Version = constants.CurrentVersion
 		cr.Spec.Version = ""
 	}
+
+	// we are under test or someone is trying to set both secret and username and password we use the username and password and skip the secret name
+	if len(cr.Spec.CommonConfig.SecretAdminCredentials) > 0 && len(cr.Spec.CommonConfig.AdminUser) > 0 && len(cr.Spec.CommonConfig.AdminPassword) > 0 {
+		log.Info("Removing SecretAdminCrdentials because username and password are provided, creating Admin Secret using those credentials")
+		cr.Spec.CommonConfig.SecretAdminCredentials = ""
+		errSecret := createSecret(service, cr.Namespace, constants.KIE_ADMIN_CREDENTIALS_SECRET, cr.Spec.CommonConfig.AdminUser, cr.Spec.CommonConfig.AdminPassword)
+		if errSecret != nil {
+			log.Error("Can't create Admin Secret. ", errSecret)
+			// here we continue instead of return otherwise fails test without k8s runtime to see the created secret
+		}
+	}
+
+	if len(cr.Spec.CommonConfig.SecretAdminCredentials) > 0 && len(cr.Spec.CommonConfig.AdminUser) == 0 && len(cr.Spec.CommonConfig.AdminPassword) == 0 {
+		errSecret, _, _ := checkAndCreateAdminSecret(service, cr.Spec.CommonConfig.SecretAdminCredentials, cr.Namespace, cr.Spec.Environment)
+		if errSecret != nil {
+			log.Error("Can't create Admin Secret. ", errSecret)
+			// here we continue instead of return otherwise fail tests without k8s runtime to see the created secret
+		}
+	}
+
+	// we create a secret with default username and password generated
+	if len(cr.Spec.CommonConfig.SecretAdminCredentials) == 0 && len(cr.Spec.CommonConfig.AdminUser) == 0 && len(cr.Spec.CommonConfig.AdminPassword) == 0 {
+		errSecret, _, _ := checkAndCreateAdminSecret(service, constants.KIE_ADMIN_CREDENTIALS_SECRET, cr.Namespace, cr.Spec.Environment)
+		if errSecret != nil {
+			log.Error("Can't create Admin Secret. ", errSecret)
+			// here we continue instead of return otherwise fails test without k8s runtime to see the created secret
+		}
+	}
+
+	// we create the kie admin secret with the provided username and password
+	if len(cr.Spec.CommonConfig.SecretAdminCredentials) == 0 && len(cr.Spec.CommonConfig.AdminUser) > 0 && len(cr.Spec.CommonConfig.AdminPassword) > 0 {
+		errSecret := createSecret(service, cr.Namespace, constants.KIE_ADMIN_CREDENTIALS_SECRET, cr.Spec.CommonConfig.AdminUser, cr.Spec.CommonConfig.AdminPassword)
+		if errSecret != nil {
+			log.Error("Can't create Admin Secret. ", errSecret)
+			// here we continue instead of return otherwise fails test without k8s runtime to see the created secret
+		}
+	}
```

Comment:
Not sure how exactly are all of these checks used and needed. Maybe it's little to complicated.
Secret should be used only in case that is set the secret name and there is no configuration for adminUser and adminPassword, right?
If there is no secret or any config for adminUser and adminPwd, then are used generated values for adminUser and adminPassword.
In that case wouldn't be easier to just check that `secret > 0 & adminUser = 0 & adminPwd = 0` ?